enterprise/backend/src/metabase_enterprise/metabot_v3/api.clj

```diff
@@ -57,6 +61,91 @@
+                             :created_at
+                             [:parent_collection.id :collection_id]
+                             [:parent_collection.name :collection_name]]
+                            {:left-join [[:report_card :card] [:and [:in :model [[:inline "dataset"] [:inline "metric"]]]  [:= :model_id :card.id]]
+                                         [:collection :collection] [:and [:= :model [:inline "collection"]]  [:= :model_id :collection.id]]
+                                         ;; TODO: How to join parent collections from the location column?
+                                         [:collection :parent_collection] [:= :card.collection_id :parent_collection.id]]
+                             :where [:= :metabot_id id]
+                             :limit limit
+                             :offset offset
+                             :order-by [[:name :asc]]})
+        total (t2/count :model/MetabotEntity :metabot_id id)]
+    {:items (for [{:keys [collection_id model] :as entity} entities]
+              (cond-> entity
+                (and (nil? collection_id)
+                     (not= :collection model)) (assoc :collection_id (:id root-collection)
+                                                      :collection_name (:name root-collection))))
+     :total total
+     :limit limit
+     :offset offset}))
+
+(api.macros/defendpoint :put "/metabots/:id/entities"
+  "Update the entities this metabot has access to"
+  [{:keys [id]} :- [:map [:id pos-int?]]
+   _query-params
+   {:keys [items]} :- [:map
+                       [:items [:sequential [:map
+                                             [:id pos-int?]
+                                             [:model [:enum "dataset" "metric" "collection"]]]]]]]
+  (api/check-superuser)
+  (api/check-404 (t2/exists? :model/Metabot :id id))
+  (t2/with-transaction [_conn]
+    (doseq [{model-id :id model :model} items]
+      (when-not (t2/exists? :model/MetabotEntity
+                            :metabot_id id
+                            :model model
+                            :model_id model-id)
+        (t2/insert! :model/MetabotEntity
+                    {:metabot_id id
+                     :model model
+                     :model_id model-id}))))
+  api/generic-204-no-content)
+
+(api.macros/defendpoint :delete ["/metabots/:id/entities/:model/:model-id" :model #"dataset|metric|collection"]
+  "Remove an entity from this metabot's access list"
+  [{:keys [id model model-id]} :- [:map
+                                   [:id pos-int?]
+                                   [:model [:enum "dataset" "metric" "collection"]]
+                                   [:model-id pos-int?]]]
+  (api/check-superuser)
+  (api/check-404 (t2/exists? :model/Metabot :id id))
+  (t2/delete! :model/MetabotEntity
+              :metabot_id id
+              :model model
+              :model_id model-id)
+  api/generic-204-no-content)
+
```

Comment:
I don't think we're completely consistent but my impression is that most of our APIs use singular... at least for the earliest entity, like `metabots` here?

Worth thinking about though it does seem like we're very inconsistent already.

Maybe also worth considering pulling out the "metabots" management into its own API namespace though, you could guard it with `metabase.api.common/+check-superuser` and then not need to worry so much about adding it to each individual endpoint?